Common/include/toolboxes/multilayer_perceptron/CIOMap.hpp

```diff
@@ -0,0 +1,190 @@
+ * definition. There are three MLP's loaded. MLP_1 predicts temperature and specific heat
+ * based on density and energy. MLP_2 predicts pressure and speed of sound based on density and
+ * energy as well. MLP_3 predicts density and energy based on pressure and temperature.
+ * During a certain look-up operation in the CFluidModel, temperature, speed of sound and pressure
+ * are needed for a given density and energy. What the CIOMap does is to point to MLP_1 for
+ * temperature evalutation, and to MLP_2 for pressure and speed of sound evaluation. MLP_3 is
+ * not considered, as the respective inputs and outputs don't match with the function call
+ * inputs and outputs.
+ *
+ *  call variables:      MLP inputs:                     MLP outputs:           call outputs:
+ *
+ *                        2--> energy --|            |--> temperature --> 1
+ *                                      |--> MLP_1 --|
+ *  1:density            1--> density --|            |--> c_p                   1:temperature
+ *  2:energy                                                                    2:speed of sound
+ *                       1--> density --|            |--> pressure --> 3        3:pressure
+ *                                      |--> MLP_2 --|
+ *                        2--> energy --|            |--> speed of sound --> 2
+ *
+ *                           pressure --|            |--> density
+ *                                      |--> MLP_3 --|
+ *                        temperature --|            |--> energy
+ *
+ *
+ * \author E.Bunschoten
+ */
+
+{
+ private:
+  std::vector<std::size_t> MLP_indices; /*!< Loaded MLP index */
+
+  std::vector<std::vector<std::pair<std::size_t, std::size_t>>>
+      Input_Map,  /*!< Mapping of call variable inputs to matching MLP inputs */
+      Output_Map; /*!< Mapping of call variable outputs to matching MLP outputs */
+
+ public:
+  /*!
+   * \brief CIOMap class constructor;
+   * \param[in] MLP_collection - Pointer to CLookUp_ANN class for which the input-output amap is to be created
+   * \param[in] inputs - Input names for the call function. These should match with at least one of the MLP inputs.
+   * \param[in] outputs - Output names for the call function. These should match with at least one of the MLP outputs.
+   */
+  CIOMap(CLookUp_ANN* MLP_collection, su2vector<std::string>& inputs, su2vector<std::string>& outputs);
+
+  /*!
+   * \brief Set MLP index in IO map
+   * \param[in] iMLP - loaded MLP index
+   */
+  void Push_MLP_index(std::size_t iMLP) { MLP_indices.push_back(iMLP); }
+
+  /*!
+  * \brief Pair call inputs and outputs with the inputs and outputs of
+          the loaded MLPs
+  * \param[in] MLP_collection - pointer to MLP collection class
+  * \param[in] inputs - vector with call input variable names
+  * \param[in] outputs - vector with call output variable names
+  */
+  void PairVariableswithMLPs(CLookUp_ANN* MLP_collection, su2vector<std::string>& inputs,
```

Comment:
Please remove the underscores in the middle of CamelCase names. It is not consistent within this class.